src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java

```
@@ -0,0 +1,62 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class UserLocationServiceImpl implements UserLocationService {
+    private final UserLocationRepository userLocationRepository;
+
+    @Autowired
+    public UserLocationServiceImpl(UserLocationRepository userLocationRepository) {
+        this.userLocationRepository = userLocationRepository;
+    }
+
+    @Override
+    public List<UserLocationDto> getAll() {
+        List<UserLocationDto> users = new ArrayList<>();
+        for (UserLocation u : userLocationRepository.findAll()) {
+            users.add(UserLocationConverter.entityToDto(u));
+        }
+        return users;
+    }
+
+    @Override
+    public UserLocationDto add(UserLocationDto userLocationDto) {
+        userLocationRepository.save(UserLocationConverter.dtoToEntity(userLocationDto));
+        return userLocationDto;
+    }
+
+    @Override
+    public UserLocationDto getById(UUID id) {
+        UserLocation userLocation = userLocationRepository.findById(id).orElseThrow();
+        return UserLocationConverter.entityToDto(userLocation);
+    }
+
+    @Override
+    public void update(UUID id, UserLocationDto userLocationDto) {
+        UserLocation userLocation = userLocationRepository.findById(id).orElseThrow();
+        //userLocation.setUserDetails(userLocationDto.getUserDetails());
+        userLocation.setCountry(userLocationDto.getCountry());
+        userLocation.setCity(userLocationDto.getCity());
+        userLocation.setAddressLine(userLocationDto.getAddressLine());
+        userLocation.setDeleted(userLocationDto.isDeleted());
+
+        userLocationRepository.save(userLocation);
+    }
+
+    @Override
+    public void toArchive(UUID id) {
+        UserLocation location = userLocationRepository.findById(id).orElseThrow();
```

fine for now but you'll need to throw an appropriate exception which you can handle in a ControllerAdvice and return a 404 error to the client

---

okey thank you